.github/workflows/s3-publish-cf-templates.yml

```diff
@@ -1,15 +1,34 @@
 name: Publish CloudFormation templates to AWS S3
 
 on:
+  push:
+    branches:
+      - main
+    paths:
+      - './packages/artillery/lib/platform/aws/iam-cf-templates/**'
```

Comment:
We'll have to change this, as the CF stack that is in main will be for an unreleased version of the CLI.

We either do `workflow_dispatch` only, but then we need to be able to specify which artillery version to test against (e.g. the latest canary), or we do `workflow_call` and trigger this for canary and for mainline.

In either case, it sounds like we might need to actually publish two versions of the templates, one for the canary, and one for the latest.